src/serialmonitor/serialMonitor.ts

```diff
@@ -38,256 +30,170 @@ export class SerialMonitor implements vscode.Disposable {
-
-    private _serialPortCtrl: SerialPortCtrl = null;
-
-    private _outputChannel: vscode.OutputChannel;
-
-    private _bufferedOutputChannel: BufferedOutputChannel;
-
-    public initialize() {
-        let defaultBaudRate;
-        if (ArduinoContext.arduinoApp && ArduinoContext.arduinoApp.settings && ArduinoContext.arduinoApp.settings.defaultBaudRate) {
-            defaultBaudRate = ArduinoContext.arduinoApp.settings.defaultBaudRate;
-        } else {
-            defaultBaudRate = SerialMonitor.DEFAULT_BAUD_RATE;
-        }
-        let defaultTimestampFormat;
-        if (ArduinoContext.arduinoApp && ArduinoContext.arduinoApp.settings && ArduinoContext.arduinoApp.settings.defaultTimestampFormat) {
-            defaultTimestampFormat = ArduinoContext.arduinoApp.settings.defaultTimestampFormat;
-        } else {
-            defaultTimestampFormat = SerialMonitor.DEFAULT_TIMESTAMP_FORMAT;
-        }
-        this._outputChannel = vscode.window.createOutputChannel(SerialMonitor.SERIAL_MONITOR);
-        this._bufferedOutputChannel = new BufferedOutputChannel(this._outputChannel.append, 300);
-        this._currentBaudRate = defaultBaudRate;
-        this._currentTimestampFormat = defaultTimestampFormat;
-        this._portsStatusBar = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Right, constants.statusBarPriority.PORT);
-        this._portsStatusBar.command = "arduino.selectSerialPort";
-        this._portsStatusBar.tooltip = "Select Serial Port";
-        this._portsStatusBar.show();
-
-        this._openPortStatusBar = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Right, constants.statusBarPriority.OPEN_PORT);
-        this._openPortStatusBar.command = "arduino.openSerialMonitor";
-        this._openPortStatusBar.text = `$(plug)`;
-        this._openPortStatusBar.tooltip = "Open Serial Monitor";
-        this._openPortStatusBar.show();
-
-        this._baudRateStatusBar = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Right, constants.statusBarPriority.BAUD_RATE);
-        this._baudRateStatusBar.command = "arduino.changeBaudRate";
-        this._baudRateStatusBar.tooltip = "Baud Rate";
-        this._baudRateStatusBar.text = defaultBaudRate.toString();
-
-        this._timestampFormatStatusBar = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Right,
+        this.timestampFormatStatusBar = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Right,
                                                                            constants.statusBarPriority.TIMESTAMP_FORMAT);
-        this._timestampFormatStatusBar.command = "arduino.changeTimestampFormat";
-        this._timestampFormatStatusBar.tooltip = `Timestamp Format: "${defaultTimestampFormat}"`;
-        this._timestampFormatStatusBar.text = `$(watch)`;
+        this.timestampFormatStatusBar.command = "arduino.changeTimestampFormat";
+        // Get the value from the serial monitor extension settings.

```

Comment:
It's not clear what this means. Is it saying we get the value later, after initialization, that we never need to get the value because the serial monitor is tracking it for us, or is it a TODO?

Reply:
We never need to get the value, I'll update the comment to be more clear.

Reply:
Actually, the comment really doesn't even need to be there anymore, but I'll make sure to comment somewhere that the serial monitor extension controls this now.